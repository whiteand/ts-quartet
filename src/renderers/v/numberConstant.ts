import { ISchemaRenderer } from "../../types";
import { getDispatchedRenderer } from "../utils";
import {
  getSimpleCondition,
  getStrictComparisonWithSchemaRenderer
} from "./utils";

const nanRenderer: ISchemaRenderer<number> = getSimpleCondition(
  valueId => `Number.isNaN(${valueId})`,
  valueId => `!Number.isNaN(${valueId})`
);

const simpleNumber: ISchemaRenderer<
  number
> = getStrictComparisonWithSchemaRenderer("validNumber");

export const numberConstantRenderer: ISchemaRenderer<
  number
> = getDispatchedRenderer((schema: number) =>
  Number.isNaN(schema) ? nanRenderer : simpleNumber
);
