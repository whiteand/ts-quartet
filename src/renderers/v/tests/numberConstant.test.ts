import { numberConstantRenderer } from "../numberConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test("v(NaN)", () => {
    snapshot(numberConstantRenderer, { schema: 0 / 0 });
  });
  test("v(42)", () => {
    snapshot(numberConstantRenderer, { schema: 42 });
  });
});
