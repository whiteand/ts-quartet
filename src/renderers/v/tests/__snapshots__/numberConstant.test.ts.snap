// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v(42) 1`] = `"_valueId === _validNumber_42"`;

exports[`v constant number renderer v(42) 2`] = `"_valueId !== _validNumber_42"`;

exports[`v constant number renderer v(42) 3`] = `"if (_valueId === _validNumber_42) return true;"`;

exports[`v constant number renderer v(42) 4`] = `"if (_valueId !== _validNumber_42) return false;"`;

exports[`v constant number renderer v(NaN) 1`] = `"Number.isNaN(_valueId)"`;

exports[`v constant number renderer v(NaN) 2`] = `"!Number.isNaN(_valueId)"`;

exports[`v constant number renderer v(NaN) 3`] = `"if (Number.isNaN(_valueId)) return true;"`;

exports[`v constant number renderer v(NaN) 4`] = `"if (!Number.isNaN(_valueId)) return false;"`;
